Mathlib/MeasureTheory/Integral/PathIntegral/Basic.lean

```diff
@@ -0,0 +1,489 @@
+then it is path integrable along any $C^1$ path in this set. -/
+theorem ContinuousOn.pathIntegrable_of_contDiffOn [NormedSpace ℝ E] {s : Set E}
+    (hω : ContinuousOn ω s) (hγ : ContDiffOn ℝ 1 γ.extend I) (hγs : ∀ t, γ t ∈ s) :
+    PathIntegrable ω γ := by
+  apply ContinuousOn.intervalIntegrable_of_Icc zero_le_one
+  simp only [funext (pathIntegralFun_def ω γ)]
+  apply ContinuousOn.clm_apply
+  · exact hω.comp (by fun_prop) fun _ _ ↦ hγs _
+  · exact hγ.continuousOn_derivWithin uniqueDiffOn_Icc_zero_one le_rfl
+
+end PathOperations
+
+/-!
+### Algebraic operations on the 1-form
+-/
+
+section Algebra
+
+variable {𝕜 E F : Type*} [RCLike 𝕜] [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+  [NormedAddCommGroup F] [NormedSpace 𝕜 F] {a b : E}
+  {ω ω₁ ω₂ : E → E →L[𝕜] F} {γ : Path a b} {t : ℝ}
+
+@[simp]
+theorem pathIntegralFun_add :
+    pathIntegralFun (ω₁ + ω₂) γ = pathIntegralFun ω₁ γ + pathIntegralFun ω₂ γ := by
+  ext; simp [pathIntegralFun]
+
+protected nonrec theorem PathIntegrable.add (h₁ : PathIntegrable ω₁ γ) (h₂ : PathIntegrable ω₂ γ) :
+    PathIntegrable (ω₁ + ω₂) γ := by
+  simpa [PathIntegrable] using h₁.add h₂
+
+theorem pathIntegral_add (h₁ : PathIntegrable ω₁ γ) (h₂ : PathIntegrable ω₂ γ) :
+    pathIntegral (ω₁ + ω₂) γ = (∫ᵖ x in γ, ω₁ x) + ∫ᵖ x in γ, ω₂ x := by
+  letI : NormedSpace ℝ F := .restrictScalars ℝ 𝕜 F
+  simp only [pathIntegral, pathIntegralFun_add]
+  exact intervalIntegral.integral_add h₁ h₂
+
+theorem pathIntegral_fun_add (h₁ : PathIntegrable ω₁ γ) (h₂ : PathIntegrable ω₂ γ) :
+    ∫ᵖ x in γ, ω₁ x + ω₂ x = (∫ᵖ x in γ, ω₁ x) + ∫ᵖ x in γ, ω₂ x :=
+  pathIntegral_add h₁ h₂
+
+@[simp]
+theorem pathIntegralFun_zero : pathIntegralFun (0 : E → E →L[𝕜] F) γ = 0 := by
+  ext; simp [pathIntegralFun]
+
+@[simp]
+theorem pathIntegralFun_fun_zero : pathIntegralFun (fun _ ↦ 0 : E → E →L[𝕜] F) γ = 0 :=
+  pathIntegralFun_zero
+
+theorem PathIntegrable.zero : PathIntegrable (0 : E → E →L[𝕜] F) γ := by
+  simp [PathIntegrable, IntervalIntegrable.zero]
+
+theorem PathIntegrable.fun_zero : PathIntegrable (fun _ ↦ 0 : E → E →L[𝕜] F) γ := .zero
+
+@[simp]
+theorem pathIntegral_zero : pathIntegral (0 : E → E →L[𝕜] F) γ = 0 := by simp [pathIntegral]
+
+@[simp]
+theorem pathIntegral_fun_zero : ∫ᵖ _ in γ, (0 : E →L[𝕜] F) = 0 := pathIntegral_zero
+
+@[simp]
+theorem pathIntegralFun_neg : pathIntegralFun (-ω) γ = -pathIntegralFun ω γ := by
+  ext; simp [pathIntegralFun]
+
+nonrec theorem PathIntegrable.neg (h : PathIntegrable ω γ) : PathIntegrable (-ω) γ := by
+  simpa [PathIntegrable] using h.neg
+
+theorem PathIntegrable.fun_neg (h : PathIntegrable ω γ) : PathIntegrable (-ω ·) γ :=
+  h.neg
+
+@[simp]
+theorem PathIntegrable.neg_iff : PathIntegrable (-ω) γ ↔ PathIntegrable ω γ :=
+  ⟨fun h ↦ by simpa using h.neg, .neg⟩
+
+@[simp]
+theorem PathIntegrable.fun_neg_iff : PathIntegrable (-ω ·) γ ↔ PathIntegrable ω γ :=
+  PathIntegrable.neg_iff
+
+@[simp]
+theorem pathIntegral_neg : pathIntegral (-ω) γ = -∫ᵖ x in γ, ω x := by
```

Comment:
use notation?

Reply:
AFAICT, if I use the notation in the LHS, then Lean will simplify using `ContinuousLinearMap.coe_neg'` first. Same in other cases. I don't know if we'll need the unapplied versions though.